odoo/osv/expression.py

```diff
@@ -361,16 +375,7 @@ def generate_table_alias(src_table_alias, joined_tables=[]):
         return '%s' % alias, '%s' % _quote(alias)
     for link in joined_tables:
         alias += '__' + link[1]
-    # Use an alternate alias scheme if length exceeds the PostgreSQL limit
-    # of 63 characters.
-    if len(alias) >= 64:
-        # We have to fit a crc32 hash and one underscore
-        # into a 63 character alias. The remaining space we can use to add
-        # a human readable prefix.
-        alias_hash = hex(crc32(alias.encode('utf-8')))[2:]
-        ALIAS_PREFIX_LENGTH = 63 - len(alias_hash) - 1
-        alias = "%s_%s" % (
-            alias[:ALIAS_PREFIX_LENGTH], alias_hash)
+    alias = _shorten_alias(alias)
     return '%s' % alias, '%s as %s' % (_quote(joined_tables[-1][0]), _quote(alias))
```

Comment:
```suggestion
    return '%s' % alias, '%s as %s' % (_quote(joined_tables[-1][0]), _quote(_shorten_alias(alias)))
```

Reply:
This change would have to be done for both "alias" references in the return clause, otherwise we'll run into 

```  File "/opt/odoo/custom/src/odoo/odoo/osv/query.py", line 126, in add_join
    assert lhs in aliases, "Left-hand-side table %s must already be part of the query tables %s!" % (lhs, str(self.tables))